presto-docs/src/main/sphinx/develop/functions.rst

```diff
@@ -2,31 +2,68 @@
 Functions
 =========
 
+Plugin Implementation
+---------------------
 The function framework is used to implement SQL functions. Presto includes a
-number of built-in functions, and an internal ``Plugin`` (plugins that have a
-dependency on presto-main) can provide new functions by returning a
-``FunctionFactory`` from ``getServices()``.
+number of built-in functions. In order to implement new functions, you can
+write a plugin that returns one more more functions from ``getFunctions()``. For example:
```

Comment:
This should be "returns one or more classes from", since a class can contain one or more functions.